pkg/updater/github.go

```diff
@@ -303,37 +301,39 @@ func (g *Github) DownloadRelease(ctx context.Context, r *github.RepositoryReleas
 	}
 	trace.EndCall(traceCtx)
 
-	file, err = g.processArchive(ctx, file, tmpDir, execName, asset)
-	return file, cleanupFn, err
+	assetPath, err = g.processArchive(ctx, assetPath, tmpDir, execName, asset)
+	return assetPath, cleanupFn, err
 }
 
 // processArchive checks if the file is an archive, and extracts it looking for the same file
 // as the current executable. If it is not an archive then the file given is returned
 func (g *Github) processArchive(ctx context.Context, file, tmpDir, execName string, asset *github.ReleaseAsset) (string, error) {
-	// TODO: better support for other archives in the future
-	if strings.HasSuffix(asset.GetName(), ".tar.gz") {
-		f, err := os.Open(file)
-		if err != nil {
-			return "", errors.Wrap(err, "failed to open downloaded archive")
-		}
-		defer f.Close()
+	if !strings.Contains(asset.GetName(), ".tar") {
+		return file, nil
+	}
 
-		storageDir := filepath.Join(tmpDir, strings.TrimSuffix(asset.GetName(), ".tar.gz"))
-		err = os.MkdirAll(storageDir, 0o755)
-		if err != nil {
-			return "", err
-		}
+	f, err := os.Open(file)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to open downloaded archive")
+	}
+	defer f.Close()
 
-		// use the name of the executable here to allow for multiple clis in a given repository
-		file, err = g.getFileFromArchive(ctx, f, storageDir, execName)
-		if err != nil {
-			return "", errors.Wrap(err, "failed to extract archive")
-		}
+	storageDir := filepath.Join(tmpDir, strings.TrimSuffix(asset.GetName(), ".tar.gz"))
```

Comment:
You check for `.tar` but expect `.tar.gz`

Reply:
If there's a reason, it needs to be a comment.

Reply:
Good catch, I just need to remove the extension here